src/streaming/multiplex/advanced.rs

```diff
@@ -304,7 +304,9 @@ impl<T> Multiplex<T> where T: Dispatch {
         while self.run {
             // TODO: Only read frames if there is available space in the frame
             // buffer
-            if let Async::Ready(frame) = try!(self.dispatch.get_mut().inner.transport().poll()) {
+            if self.frame_buf.full() {
```

Comment:
I'll have to figure out how to write a small test that can trigger a filled buffer case consistently. I was able to trigger it on a large project of mine but that doesn't readily translate to a good unit test.